pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +28,123 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:scope=Namespaced
 type StorageCohort struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a cohort.
+	// +required
+	Spec StorageCohortSpec `json:"spec"`
+
+	// Most recently observed status of the cohort.
+	// Populated by the cohort operator or cohort manager.
+	// +optional
+	Status StorageCohortStatus `json:"status,omitempty"`
 }
 
+// StorageCohortSpec describes the attributes that a cohort is created with.
+type StorageCohortSpec struct {
+	// NodeSelector is used to specify the cohort to be considered
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+	// CohortManager stores all the details about the cohort manager responsible for
+	// managing the cohort
+	// +optional
+	CohortManager interface{} `json:"cohortManager,omitempty"`
+
+	// DefaultStorageProvisioner is the default provisioner for the cohort which can be used
+	// used for provisioning pools or volumes when no provisioner is specified in the storage pool
+	// For example: "openebs.io/scp-lvm-provisioner" or "openebs.io/scp-device-provisioner".
+	// +optional
+	DefaultStorageProvisioner string `json:"defaultStorageProvisioner,omitempty"`
+}
+
+// StorageCohortStatus stores information about the current status of a storage cohort.
+type StorageCohortStatus struct {
+	// Components is an array of different component's conditions which the cohort is comprised of.
+	// +optional
+	Components []ComponentStatus `json:"components,omitempty"`
+
+	// Capabilities represent capabilities that a cohort consists of
+	// +optional
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+}
+
+// ComponentStatus stores information about the current status of storage cohort's components.
+type ComponentStatus struct {
```

Comment:
can we add more details on how each condition having impact in different use case.
One example can be for scheduling which condition we need to look. As comments will be used as godoc

Reply:
For now, only cohort condition are well defined and rest are left to the users to implement. So, will add for the cohort only.

Reply:
ok